python/pyarrow/_fs.pyx

```diff
@@ -496,6 +678,24 @@ cdef class FileSystem(_Weakrefable):
         missing_dir_ok : boolean, default False
             If False then an error is raised if path does
             not exist
+
+        Examples
+        --------
+        Create a directory and copy a file into it:
+
+        >>> local.create_dir(local_path + '/new_dir')
+        >>> local.copy_file(path,
+        ...                 local_path + '/new_dir/example_copy.dat')
+        >>> local.get_file_info(local_path + '/new_dir/example_copy.dat')
+        <FileInfo for '/.../new_dir/example_copy.dat': type=FileType.File, size=4>
+
+        Delete only the content of the directory and not the directory itself:
+
+        >>> local.delete_dir_contents(local_path + '/new_dir')
+        >>> local.get_file_info(local_path + '/new_dir/example_copy.dat')
+        <FileInfo for '/.../new_dir/example_copy.dat': type=FileType.NotFound>
+        >>> local.get_file_info(local_path + '/new_dir')
+        <FileInfo for '/.../new_dir': type=FileType.Directory>
```

Comment:
The fact that the example is much longer and more complex than the method description makes it not very useful IMHO.